keycloak/sms-provider/src/main/java/org/sunbird/keycloak/resetcredential/sms/KeycloakSmsAuthenticator.java

```diff
@@ -66,13 +72,19 @@ public void authenticate(AuthenticationFlowContext context) {
         }
 
         if (StringUtils.isNotBlank(mobileNumber) || StringUtils.isNotBlank(userEmail)) {
-          if (StringUtils.isNotBlank(userEmail)) {
+          if (StringUtils.isBlank(mobileNumber) && StringUtils.isNotBlank(userEmail)) {
```

Comment:
StringUtils.isNotBlank(userEmail)  -> not required to check - based on previous condition -> if mobile is blank - email will not be blank